src/data-access/typeORMDataAccess.ts

```diff
@@ -1,254 +1,208 @@
+      return results;
+    }
+  );
+  return res;
 }
 
 export async function getGroups() {
-  try {
-    const res = await AppDataSource.transaction(
-      async (transactionalEntityManager) => {
-        const results = await transactionalEntityManager
-          .getRepository(Group)
-          .createQueryBuilder('group')
-          .orderBy('group.name', 'ASC')
-          .getMany();
-        return results;
-      }
-    );
-    return res;
-  } catch (error) {
-    console.error('Database error:', error);
-    throw error;
-  }
+  const res = await AppDataSource.transaction(
+    async (transactionalEntityManager) => {
+      const results = await transactionalEntityManager
+        .getRepository(Group)
+        .createQueryBuilder('group')
+        .orderBy('group.name', 'ASC')
+        .getMany();
+      return results;
+    }
+  );
+  return res;
 }
 
 export async function updateGroup(id: string, groupUpdates: GroupUpdates) {
-  try {
-    const res = await AppDataSource.transaction(
-      async (transactionalEntityManager) => {
-        let results = await transactionalEntityManager
-          .getRepository(Group)
-          .findOneBy({
-            id,
-          });
-        if (results) {
-          await transactionalEntityManager
-            .getRepository(Group)
-            .merge(results, groupUpdates);
-          results = await transactionalEntityManager
-            .getRepository(Group)
-            .save(results);
-        }
-        return results;
-      }
-    );
-    return res;
-  } catch (error) {
-    console.error('Database error:', error);
-    throw error;
-  }
+  const res = await AppDataSource.transaction(
+    async (transactionalEntityManager) => {
+      let results = await transactionalEntityManager
+        .getRepository(Group)
+        .findOneBy({
+          id,
+        });
+      if (!results) throw new FindGroupError(`no group with id: ${id}`);
+      transactionalEntityManager
+        .getRepository(Group)
+        .merge(results, groupUpdates);
+      results = await transactionalEntityManager
+        .getRepository(Group)
+        .save(results);
+      return results;
+    }
+  );
+  return res;
 }
 
 export async function deleteGroup(id: string) {
-  try {
-    const res = await AppDataSource.transaction(
-      async (transactionalEntityManager) => {
-        const results = await transactionalEntityManager
-          .getRepository(Group)
+  const res = await AppDataSource.transaction(
+    async (transactionalEntityManager) => {
+      const results = await transactionalEntityManager
+        .getRepository(Group)
+        .findOneBy({
+          id,
+        });
+      if (!results) throw new FindGroupError(`no group with id: ${id}`);
+      await transactionalEntityManager.getRepository(Group).remove(results);
+      return results;
+    }
+  );
+  return res;
+}
+
+export async function addUsersToGroup(groupId: string, userIds: string[]) {
```

Comment:
Read [this article](https://github.com/typeorm/typeorm/blob/master/docs/relational-query-builder.md), maybe it will give you a hint on how to simplify your solution a bit.

Reply:
fixed, remain previous version committed